Tc dp delete item

tcelovsky opened the pull request:
## Description

We added delete an item functionality, added `.trim()` to the user token value to get rid of any extra spaces and added a short welcome message to the home page. 

<!-- What does this code change? Why did I choose this approach? Did I learn anything worth sharing? Reminder: This will be a publicly facing representation of your work (READ: help you land that sweet dev gig). -->

## Related Issue

Closes #11 
<!-- If you write "closes" followed by the Github issue number, it will automatically close the issue for you when the PR merges -->

## [Acceptance Criteria](https://github.com/the-collab-lab/tcl-27-smart-shopping-list/issues/11)

- User is able to delete an item from the shopping list
- Before deleting, prompt the user to confirm that they really want to delete the item to prevent accidental deletions
- Deletion should cause the associated record(s) in the database to be deleted
<!-- Include AC from the Github issue -->

## Type of Changes

<!-- Put an `✓` for the applicable box: -->

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria

<!-- Provide steps the other cohort members and mentors need to follow to properly test your additions. -->
- From your terminal, pull down this branch with `git pull origin tc-dp-delete-item` and check that branch out with `git checkout tc-dp-delete-item`.
- First log in or create a new list and add items to that list.
- Click "X" button next to the item name.
- When prompted, click "Cancel" and confirm that the item was **not** deleted.
- Click "X" button next to the item name again and when prompted, click "Yes".
- Ensure that the item disappears from the screen and double check that it is deleted from the [database](https://console.firebase.google.com/u/0/project/tcl-27-shopping-list/firestore/data/~2Fgroceries).

tcelovsky committed:
Co-authored-by: Dani Poplin <danih37@gmail.com>

Add delete item functionality.

tcelovsky committed:
Co-authored-by: Dani Poplin <danih37@gmail.com>

Add a welcome message to the home screen; trim token value of extra spaces.

YolandaHaynes reviewed (approved):
I like the trim idea! It really helps out making sure the token passes in case someone puts in an extra space before/after the token.

My only issues I had was when I tried to manually type in the token, it won't allow me to input empty space in between the words. I tried in a different browser with same result. I understand trim works by removing space before and after a string, so I am not sure why I can't add a space between each word. However, I can copy and paste token with no issue! Not sure if that's on my end or if anyone else is having that issue. 

Other than that, everything works on my end. I can delete items, there was a prompt and it did remove that item from the database.

AirZT11 commented on src/components/containers/Home.js:
```diff
@@ -30,7 +30,7 @@ class Home extends Component {
   handleChange = (e) => {
     this.setState({
-      token: e.target.value,
+      token: e.target.value.trim(),
```
Like Yolanda previously mentioned, I can't type in any spaces when typing into the Home page token field. I believe it's because each letter typed is a string within itself so the trim() method is called on every input, which doesn't allow any whitespace before or after. 

A way around this is to add the trim() method in the handleClick() function to trim() the token that's been placed in state:
```
handleClick = (e) => {
    e.preventDefault();

    /* FIRST trim() USED TO FIND MATCHING TOKEN IN TOKENS STATE */
    if (this.state.tokens.includes(this.state.token.trim())) {

      /* SECOND trim() USED TO FETCH CORRECT ITEMS FOR TOKEN IN LOCALSTORAGE */
      localStorage.setItem('token', this.state.token.trim());

      this.props.setLoggedIn(true);
      this.setState({
        token: '',
      });
    } else {
      alert('Sorry, token not found. Try again or create a new list.');
    }
 };
```

tcelovsky replied:
Thank you both @AirZT11 and @YolandaHaynes for catching this! I should have tried typing in a user token and not just copy/pasting. I updated to move trim() from handleChange to handleClick, thank you!

AirZT11 commented on src/components/GroceryCard.js:
```diff
@@ -39,6 +39,12 @@ const GroceryCard = ({ item }) => {
     }
   };
 
+  const handleDelete = () => {
```
Nice! I love the confirmation conditional logic! Great job on this!

AirZT11 reviewed (changes requested):
Great job on the delete item feature! I really like the confirmation message to validate a delete. 

I would just check back on the trim() feature in the Home container. Thanks!!

tcelovsky committed:
Co-authored-by: Dani Poplin <danih37@gmail.com>

Move trim() to handleClick from handleChange to allow a user to manually type in a user token with spaces

grellyd reviewed (approved):
Nice! Looks solid. I can confirm the acceptance criteria have been met.
